src/CSModule.sol

```diff
@@ -721,7 +726,27 @@ contract CSModule is
             incrementNonceIfUpdated: true
         });
         // Direct call of `enqueueNodeOperatorKeys` if depositable is not changed
-        depositQueue.enqueueNodeOperatorKeys(_nodeOperators, nodeOperatorId);
+        _enqueueNodeOperatorKeys(nodeOperatorId);
+    }
+
+    function _enqueueNodeOperatorKeys(uint256 nodeOperatorId) internal {
+        uint256 curveId = accounting.getBondCurveId(nodeOperatorId);
+        NodeOperator storage no = _nodeOperators[nodeOperatorId];
+
+        if (PARAMETERS_REGISTRY.isEligibleForPriorityQueue(curveId) &&
+            PARAMETERS_REGISTRY.maxKeysPerOperatorInPriorityQueue() > no.totalAddedKeys) {
```

Comment:
This condition seems off. I think we need to have {queuePriority, maxKeysInPriotiryQueue} and place up to maxKeysInPriotiryQueue to the queue with queuePriority the rest to the normal queue

Reply:
Sorry, I didn't get it. Can you give an example?

Reply:
NO has 100 keys waiting to be placed in the queue. This is a new NO and they had never uploaded keys before. queuePriority = 5, maxKeysInPriotiryQueue = 20. We need to place 20 keys to the queue with queuePriority = 5 and the rest to the default queue

Reply:
The other example. NO already has 10 keys added and enqueued in the past and 30 more waiting to be enqueued. queuePriority = 5, maxKeysInPriotiryQueue = 20. We need to add 10 keys to the queue with queuePriority = 5 and the rest to the default queue